src/rpm/builder.rs

```diff
@@ -746,11 +747,20 @@ impl RPMBuilder {
                 offset,
                 IndexData::I18NString(vec![self.desc]),
             ),
-            IndexEntry::new(
-                IndexTag::RPMTAG_LONGSIZE,
-                offset,
-                IndexData::Int64(vec![combined_file_sizes]),
-            ),
+            match combined_file_sizes.try_into() {
```

Comment:
Does this actually build?  Interesting

I would prefer something like this, having already done the calculation

```
if small_package {
              IndexEntry::new(
                  IndexTag::RPMTAG_SIZE,
                offset,
                IndexData::Int64(vec![combined_file_sizes]),
                    IndexEntry::new(IndexTag::RPMTAG_SIZE, offset, IndexData::Int32(vec![size]))
            ),
} else {
              IndexEntry::new(
                  IndexTag::RPMTAG_LONGSIZE,
                  offset,
                  IndexData::Int64(vec![combined_file_sizes]),
              )
}